test/liquidity-protection/LiquidityProtection.js

```diff
@@ -2325,6 +2338,277 @@ describe('LiquidityProtection', () => {
+                            await liquidityProtection.setTotalPositionsValue(poolToken.address, 0);
+                            await liquidityProtection.migrateSystemPoolTokens([poolToken.address]);
+                            expect(await systemPoolTokensBalance()).equals(0);
+                        });
+
+                        it('should not migrate any system pool token when total positions value equals total liquidity', async () => {
+                            const provider = owner;
+                            await initPool(isEthReserve);
+
+                            const poolTokenSupply = await poolToken.totalSupply();
+                            await addV2Liquidity(provider, depositAmount, isEthReserve);
+                            const newPoolTokenSupply = await poolToken.totalSupply();
+                            const poolTokenSupplyDelta = newPoolTokenSupply.sub(poolTokenSupply)
+                            const totalLiquidity = await reserveTokenBalance();
+                            expect(totalLiquidity).equal(RESERVE1_AMOUNT.add(depositAmount));
+                            expect(await systemPoolTokensBalance()).equals(poolTokenSupplyDelta.div(2));
+                            
+                            await liquidityProtection.setTotalPositionsValue(poolToken.address, totalLiquidity);
+                            await liquidityProtection.migrateSystemPoolTokens([poolToken.address]);
+                            expect(await systemPoolTokensBalance()).equals(poolTokenSupplyDelta.div(2));
+                        });
+
+                        it('should not migrate any system pool token when total positions value is greater than total liquidity', async () => {
+                            const provider = owner;
+                            await initPool(isEthReserve);
+                            
+                            const poolTokenSupply = await poolToken.totalSupply();
+                            await addV2Liquidity(provider, depositAmount, isEthReserve);
+                            const newPoolTokenSupply = await poolToken.totalSupply();
+                            const poolTokenSupplyDelta = newPoolTokenSupply.sub(poolTokenSupply)
+                            const totalLiquidity = await reserveTokenBalance();
+                            expect(totalLiquidity).equal(RESERVE1_AMOUNT.add(depositAmount));
+                            expect(await systemPoolTokensBalance()).equals(poolTokenSupplyDelta.div(2));
+
+                            await liquidityProtection.setTotalPositionsValue(poolToken.address, totalLiquidity.mul(2));
+                            await liquidityProtection.migrateSystemPoolTokens([poolToken.address]);
+                            expect(await systemPoolTokensBalance()).equals(poolTokenSupplyDelta.div(2));
+                        });
+
+                        context('when the total positions value equals 80% of the total protected liquidity', async () => {
```

Comment:
I would add a few more test cases, such as:
1. when the total positions value equals 100% of the total protected liquidity
2. when the total positions value equals 120% of the total protected liquidity
3. when the total positions value equals 0 and the total protected liquidity is larger than 0
4. when the total positions value is larger than 0 and the total protected liquidity equals 0

Reply:
I might be missing something but I believe these tests already exist - can you be more specific?

Reply:
The `context` has probably confused me.
Probably better to place that (single) test on the same level along with the other tests (i.e., outside of that context).

Reply:
it's there to make the case clearer, otherwise the test description would be very long